hack/manifest-gen/main.go

```diff
@@ -0,0 +1,145 @@
+
+package main
+
+import (
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/elastic/cloud-on-k8s/hack/manifest-gen/internal"
+	"github.com/spf13/cobra"
+)
+
+var (
+	generateFlags = internal.GenerateFlags{}
+	optionsFlags  = internal.OptionsFlags{}
+	sourceFlag    string
+)
+
+func main() {
+	cmd := buildCmd()
+	if err := cmd.Execute(); err != nil {
+		log.Printf("Error: %v", err)
+		os.Exit(1)
+	}
+}
+
+func buildCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:           "manifest-gen",
+		Short:         "ECK manifest generator",
+		Long:          `Utilities for generating an installation manifest for ECK`,
+		SilenceErrors: true,
+		SilenceUsage:  true,
+	}
+
+	sourceDir, err := findSourceDir()
+	if err != nil {
+		log.Fatalf("Failed to locate assets: %v", err)
+	}
+
+	cmd.PersistentFlags().StringVar(&sourceFlag, "source", sourceDir, "Source directory")
+	_ = cmd.PersistentFlags().MarkHidden("source")
+
+	cmd.AddCommand(generateCmd())
+	cmd.AddCommand(optionsCmd())
+
+	return cmd
+}
+
+func findSourceDir() (string, error) {
+	execPath, err := os.Executable()
+	if err != nil {
+		return "", err
+	}
+
+	path, err := filepath.Abs(execPath)
+	if err != nil {
+		return "", err
+	}
+
+	return filepath.Join(filepath.Dir(path), "assets", "charts", "eck"), nil
+}
+
+func generateCmd() *cobra.Command {
+	desc := `
+Generates a manifest for installing ECK.
+
+There are two pre-defined profiles for installing ECK. The "global" profile installs ECK with access to the
+whole Kubernetes cluster. The "restricted" profile installs ECK restricted to a single (or several) namespace(s).
+
+The generated manifest can be customized using the "--set" and "--values" flags. Use the "options" command to list all
+available configuration options. The behaviour of these flags is identical to the similarly named flags used by Helm.
+See https://helm.sh/docs/intro/using_helm/ for more information.
+
+By default, the operator is installed into the "elastic-system" namespace. This can be overridden by setting the
+"operator.namespace" option.
+
+`
+	examples := `
+Global operator:
+    $ manifest-gen generate
+
+Global operator with the validation webhook disabled:
+    $ manifest-gen generate --set=config.webhook.enabled=false
+
+Global operator with resource memory limit increased to 300Mi and CPU limit increased to 2:
+    $ manifest-gen generate --set=operator.resources.limits.cpu=2,operator.resources.limits.memory=300Mi
+
+Restricted operator without CRDs, managing the "elastic-system" namespace:
+    $ manifest-gen generate --profile=restricted --exclude-crds
+
+Restricted operator installed to and managing the single namespace named "namespacex":
+    $ manifest-gen generate --profile=restricted --set=operator.namespace=namespacex --set=config.managedNamespaces='{namespacex}'
+
+Restricted operator managing "elastic-system", "nsa" and "nsb":
+    $ manifest-gen generate --profile=restricted --set=config.managedNamespaces='{elastic-system, nsa, nsb}'
```

Comment:
is it helm convention use `{a, b, c}` instead of `[a, b, c]` for lists?

Reply:
yes, https://helm.sh/docs/intro/using_helm/#the-format-and-limitations-of---set